### ISSUE TITLE
`require-optimization` should ignore components not extending `React.Component`

### ISSUE THREAD

**Issue by christianharke** [yannickcr/eslint-plugin-react#986]
We use a base class to extend our Components, so we do not have to optimize each Component on its own:

```javascript
import React from 'react';
import shouldPureComponentUpdate from 'react-pure-render/function';

// Optimized base class:
export default class Component extends React.Component {

    shouldComponentUpdate(...args) {
        return Reflect.apply(shouldPureComponentUpdate, this, args);
    }

}
```

However, `require-optimization` is complaining that the Components extending this base class were not optimized.

**Comment by ljharb** [comment]
Using a base class is a bad practice (article by the React team: https://facebook.github.io/react/docs/composition-vs-inheritance.html) - however, this plugin supports a JSDoc-style annotation you can add in a comment above all of your components to indicate they are react components.

What I'd recommend instead is importing `shouldPureComponentUpdate` in every single component rather than using inheritance to solve that problem (or using a decorator).

**Comment by jpickwell** [comment]
@christianharke, the plugin doesn't look for `React.Component` or `React.PureComponent`, it looks for `Component` and `PureComponent`. Notice the lack of the `React.` prefix. You could rename your base `Component` class to `BaseComponent` or anything that isn't `Component` or `PureComponent` and this would stop the plugin from treating the class like a React component.

**Comment by jpickwell** [comment]
Looking at the source, the `React.` prefix (`React` being the `pragma`) is treated as optional. So, 
it looks for all of the following:

* `Component`
* `React.Component`
* `PureComponent`
* `React.PureComponent`

(assuming `pragma` is set to the default of `React`)

**Comment by christianharke** [comment]
Thank you @jpickwell!

Just renaming the variable from `Compontent` to `BaseComponent` while importing fixed the issue.
